### PR TITLE
Aruba ClearPass - add format

### PR DESCRIPTION
**Pull request by lvoloshyn-sekoia**
## Type of Change

- [X] New intake format
- [ ] Update to existing intake format
- [ ] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product:

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Add an Aruba ClearPass parser for ingesting and categorizing network access control events.

New Features:
- Add the Aruba ClearPass intake format with support for Session Logs, Insight Logs, Audit Records, and ClearPass System Events.
- Map ClearPass event data to common observability fields and Aruba-specific endpoint, enforcement, and posture attributes.

Documentation:
- Add initial release metadata and changelog information for the Aruba ClearPass parser.

Tests:
- Add representative parser test cases covering session, insight, audit, and system event logs.

### CODE REVIEW

**Comment by mchupeau-sk** [comment]
Hey @lvoloshyn-sekoia,
### 1. `event.category` / `event.type` only set for Session Logs

Only the `set_session_fields` stage assigns `event.category` / `event.type`.
Insight Logs, Audit Records, and ClearPass System Events produce events with **no
ECS categorization**, which hurts correlation and rule matching. (Confirmed in
`test_insight_logs_1.json`, `test_audit_records_1.json`,
`test_system_events_1.json` — none contain `event.category`/`event.type`.)

**Proposed fix** — in `Aruba Network/aruba-clearpass/ingest/parser.yml`, add
categorization to the remaining stages (values below are reasonable defaults —
please confirm against the taxonomy category/type compatibility validator and
update the corresponding test `expected` blocks):

```yaml
  set_insight_fields:
    actions:
      - set:
          event.category: ["host"]
          event.type: ["info"]
      - set:
          host.type: "{{parsed_event.message.get('Endpoint.Device-Category')}}"
          # ...existing sets...

  set_audit_records_fields:
    actions:
      - set:
          event.category: ["configuration"]
          event.type: ["change"]
      - set:
          file.name: "{{parsed_event.message.fname}}"

  set_system_fields:
    actions:
      - set:
          event.category: ["process"]
          event.type: ["info"]
      - set:
          event.outcome: "{{parsed_event.message.outcome | lower}}"
          process.name: "{{parsed_event.message.deviceProcessName}}"
```

Each corresponding `tests/*.json` `expected` block must then be updated to
include the new `event.category`/`event.type` arrays, and `mise run test
--format=aruba-clearpass` + `mise run validate` re-run to confirm the
category/type pairs are accepted. *(Not applied/verified in this review — it
requires the author's taxonomy choice.)*

### 2. Empty CHANGELOG entry

`CHANGELOG.md` contains only the template with an empty `## [Unreleased]`
section — no entry describing the new parser.

**Proposed fix** — in `Aruba Network/aruba-clearpass/CHANGELOG.md`:

```diff
 ## [Unreleased]
+
+### Added
+
+- Initial version of the Aruba ClearPass parser (Session Logs, Insight Logs,
+  Audit Records, ClearPass System Events)
```

### 3. Test IPs are not TEST-NET ranges

Tests use `1.1.1.1` and `2.2.2.2`, which are real, routable public IPs
(Cloudflare / Orange). The anonymization validator did not reject them, but the
contribution guidelines require documentation/test IPs to use TEST-NET ranges.

**Proposed fix** — across `tests/*.json`, replace in both `input.message` and
`expected` (and `related.ip`, `observer.ip`, `source.ip`):

```diff
- dvc=1.1.1.1
+ dvc=192.0.2.10
- src=2.2.2.2
+ src=198.51.100.20
```

### 4. Questionable `host.*` semantic mappings (Insight Logs)

In `set_insight_fields`, `Endpoint.Device-Name` → `host.os.full` and
`Endpoint.Device-Category` → `host.type`. ECS `host.os.full` is meant for the OS
name/version and `host.type` for values like `physical`/`virtual`. The device
category/name are already captured in `aruba.clearpass.endpoint.*`, so these ECS
mappings may be misleading.

**Proposed fix** — in `Aruba Network/aruba-clearpass/ingest/parser.yml`,
consider dropping the `host.os.full`/`host.type` assignments (the data is already
in `aruba.clearpass.endpoint.device_name` / `device_category`), or remap
`host.type` only when the value is a genuine ECS host type. *(Judgment call for
the author.)*

### 5. `test_insight_logs_2.json` has no `@timestamp`

The source line has no `rt=` field, so no `@timestamp` is produced (falls back to
intake time). This is acceptable if ClearPass genuinely omits `rt` for some
Insight events, but worth confirming a timestamp source exists for that log type.

---

**Comment by mchupeau-sk** [comment]
> Hey @lvoloshyn-sekoia,
> 
> ### 1. `event.category` / `event.type` only set for Session Logs
> Only the `set_session_fields` stage assigns `event.category` / `event.type`. Insight Logs, Audit Records, and ClearPass System Events produce events with **no ECS categorization**, which hurts correlation and rule matching. (Confirmed in `test_insight_logs_1.json`, `test_audit_records_1.json`, `test_system_events_1.json` — none contain `event.category`/`event.type`.)
> 
> **Proposed fix** — in `Aruba Network/aruba-clearpass/ingest/parser.yml`, add categorization to the remaining stages (values below are reasonable defaults — please confirm against the taxonomy category/type compatibility validator and update the corresponding test `expected` blocks):
> 
> ```yaml
>   set_insight_fields:
>     actions:
>       - set:
>           event.category: ["host"]
>           event.type: ["info"]
>       - set:
>           host.type: "{{parsed_event.message.get('Endpoint.Device-Category')}}"
>           # ...existing sets...
> 
>   set_audit_records_fields:
>     actions:
>       - set:
>           event.category: ["configuration"]
>           event.type: ["change"]
>       - set:
>           file.name: "{{parsed_event.message.fname}}"
> 
>   set_system_fields:
>     actions:
>       - set:
>           event.category: ["process"]
>           event.type: ["info"]
>       - set:
>           event.outcome: "{{parsed_event.message.outcome | lower}}"
>           process.name: "{{parsed_event.message.deviceProcessName}}"
> ```
> 
> Each corresponding `tests/*.json` `expected` block must then be updated to include the new `event.category`/`event.type` arrays, and `mise run test --format=aruba-clearpass` + `mise run validate` re-run to confirm the category/type pairs are accepted. _(Not applied/verified in this review — it requires the author's taxonomy choice.)_
> 
> ### 2. Empty CHANGELOG entry
> `CHANGELOG.md` contains only the template with an empty `## [Unreleased]` section — no entry describing the new parser.
> 
> **Proposed fix** — in `Aruba Network/aruba-clearpass/CHANGELOG.md`:
> 
> ```diff
>  ## [Unreleased]
> +
> +### Added
> +
> +- Initial version of the Aruba ClearPass parser (Session Logs, Insight Logs,
> +  Audit Records, ClearPass System Events)
> ```
> 
> ### 3. Test IPs are not TEST-NET ranges
> Tests use `1.1.1.1` and `2.2.2.2`, which are real, routable public IPs (Cloudflare / Orange). The anonymization validator did not reject them, but the contribution guidelines require documentation/test IPs to use TEST-NET ranges.
> 
> **Proposed fix** — across `tests/*.json`, replace in both `input.message` and `expected` (and `related.ip`, `observer.ip`, `source.ip`):
> 
> ```diff
> - dvc=1.1.1.1
> + dvc=192.0.2.10
> - src=2.2.2.2
> + src=198.51.100.20
> ```
> 
> ### 4. Questionable `host.*` semantic mappings (Insight Logs)
> In `set_insight_fields`, `Endpoint.Device-Name` → `host.os.full` and `Endpoint.Device-Category` → `host.type`. ECS `host.os.full` is meant for the OS name/version and `host.type` for values like `physical`/`virtual`. The device category/name are already captured in `aruba.clearpass.endpoint.*`, so these ECS mappings may be misleading.
> 
> **Proposed fix** — in `Aruba Network/aruba-clearpass/ingest/parser.yml`, consider dropping the `host.os.full`/`host.type` assignments (the data is already in `aruba.clearpass.endpoint.device_name` / `device_category`), or remap `host.type` only when the value is a genuine ECS host type. _(Judgment call for the author.)_
> 
> ### 5. `test_insight_logs_2.json` has no `@timestamp`
> The source line has no `rt=` field, so no `@timestamp` is produced (falls back to intake time). This is acceptable if ClearPass genuinely omits `rt` for some Insight events, but worth confirming a timestamp source exists for that log type.

@copilot+claude-opus-5:high read the comments and try to improve the parser with the comment

File: Aruba Network/aruba-clearpass/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  - name: set_system_fields
+    filter: "{{parsed_event.message.cat == 'ClearPassSystemEvents'}}"
+
+stages:
+  set_common_fields:
+    actions:
+      - set:
+          observer.vendor: "Aruba Networks"
+          observer.product: "ClearPass"
+          event.dataset: "aruba-clearpass"
```
Comment (squioc) [inline Aruba Network/aruba-clearpass/ingest/parser.yml:27]: As we already have the information about the product with `observer.product`,
should `event.dataset` reflects the category of the event instead?

File: Aruba Network/aruba-clearpass/tests/test_audit_records_1.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "@timestamp": "2016-08-01T11:16:32Z",
+    "aruba": {
+      "clearpass": {
+        "category": "Audit Records",
+        "name": "Syslog Export Data"
+      }
+    },
+    "file": {
+      "name": "Intel Radius Authenication"
```
Comment (squioc) [inline Aruba Network/aruba-clearpass/tests/test_audit_records_1.json:27]: Is it really a filename 😒 ?

PS: I know that it copied from CEF's `fname`, but, I wonder if `file.name` is the correct field to hold this information :/

**Review by squioc — APPROVED** [review]
Hello,

LGTM globally. I made some comments however.

Thank you
